sdk/src/user.ts

```diff
@@ -485,62 +801,128 @@ export class User {
 	 */
 	public getFreeCollateral(
 		marginCategory: MarginCategory = 'Initial',
-		enterHighLeverageMode = false
+		enterHighLeverageMode = false,
+		perpMarketIndex?: number
 	): BN {
 		const totalCollateral = this.getTotalCollateral(marginCategory, true);
```

Comment:
this also needs to be using the margin type?

Reply:
As in you think maybe it doesn't need to be using that?

Reply:
like we don't need to pass `perpMarketIndex ? 'Isolated' : undefined,`

Reply:
I think in line with your other comment, we would just keep perpMarketIndex and only use that, not add a marginType parameter which would be redundant.